### PR TITLE
Use new codebase

### PR DESCRIPTION
**Pull request by hexonaut**
Updating to `sparklend-v1-core`. Also removed Goerli as the network is no longer running and throwing errors. Also removed the old DAI IRM as we don't use it anymore. Some other refresh stuff like contract verification inside the config file too.

### CODE REVIEW

File: script/DeploySpark.s.sol [hunk 1]
```diff
@@ -110,7 +110,6 @@ contract DeploySpark is Script {
         // 4. Deploy and configure ACL manager
 
         aclManager = new ACLManager(poolAddressesProvider);
-        aclManager.addPoolAdmin(deployer);
```
Comment (lucas-manuel) [inline script/DeploySpark.s.sol:113]: Why are we removing these lines?

File: script/DeploySparkConfigEngine.s.sol [hunk 2]
```diff
@@ -8,6 +8,7 @@ import {ScriptTools} from "dss-test/ScriptTools.sol";
 import {AaveV3ConfigEngine}    from 'aave-helpers/v3-config-engine/AaveV3ConfigEngine.sol';
 import {V3RateStrategyFactory} from 'aave-helpers/v3-config-engine/V3RateStrategyFactory.sol';
 
+// Note: we need to link to aave-v3-core instead of sparklend-v1-core because of conflicts with the helper library
```
Comment (lucas-manuel) [inline script/DeploySparkConfigEngine.s.sol:11]: NOTE: We

File: test/deploys/1/primary/SparkDeploy_20240408_EthereumPrimary_PoolImplementation.t.sol [hunk 3]
```diff
@@ -3,13 +3,13 @@ pragma solidity ^0.8.0;
 
 import "../../../SparkDeployPoolImplementationBase.t.sol";
 
-contract SparkDeploy_20230711_EthereumPrimary_PoolImplementationV302Test is SparkDeployPoolImplementationBaseTest {
+contract SparkDeploy_20240408_EthereumPrimary_PoolImplementationTest is SparkDeployPoolImplementationBaseTest {
     
     constructor() {
         rpcUrl      = getChain("mainnet").rpcUrl;
-        forkBlock   = 17689592;
+        forkBlock   = 19609506;
```
Comment (lucas-manuel) [inline test/deploys/1/primary/SparkDeploy_20240408_EthereumPrimary_PoolImplementation.t.sol:10]: Add date

File: script/output/100/broadcast/primary-pool-20240408.json [hunk 4]
```diff
@@ -1,169 +1,169 @@
 {
   "transactions": [
     {
-      "hash": "0x21ff37f75e6b8242887e187cfb48b9953877ea54b35be5980d760769510d7609",
+      "hash": "0xbc2b95fc8be55b269503886d893f19ebba5fed67e21a296487de5921f8abc59b",
```
Comment (lucas-manuel) [inline script/output/100/broadcast/primary-pool-20240408.json:4]: Is this just showing up as a diff cause git thinks you moved the file?

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

File: script/DeploySpark.s.sol (hunk 1 above)
Comment (hexonaut) [reply script/DeploySpark.s.sol:113]: It's not used anymore. We needed it when we were setting up reserves during the deploy, but now it just adds and removes it later so no point.

File: script/DeploySparkConfigEngine.s.sol (hunk 2 above)
Comment (hexonaut) [reply script/DeploySparkConfigEngine.s.sol:11]: fixed

File: test/deploys/1/primary/SparkDeploy_20240408_EthereumPrimary_PoolImplementation.t.sol (hunk 3 above)
Comment (hexonaut) [reply test/deploys/1/primary/SparkDeploy_20240408_EthereumPrimary_PoolImplementation.t.sol:10]: added

File: script/output/100/broadcast/primary-pool-20240408.json (hunk 4 above)
Comment (hexonaut) [reply script/output/100/broadcast/primary-pool-20240408.json:4]: yeah

**Review by barrutko — APPROVED** [review]

**Review by lucas-manuel — APPROVED** [review]

**Pull request merged**